frontend/src/components/notifications/NotificationsTab.vue

```diff
@@ -1 +1,236 @@
+
+export default {
+    data() {
+        return {
+            notifications: [],
+            checkBoxToggleAllState: false, // on/off state for toggle all checkbox
+            checkBoxListState: [], // on/off state for checkboxes in list
+            rowClickedIndex: null,
+            checkBoxClickedState: false, // on/off state for a checkbox clicked
+            message: null,
+            headers: [
+                { title: null, align: 'start', key: 'name', sortable: false, width: '8%' },
+                { title: 'Read/Unread', align: 'start', key: 'isRead', sortable: true, width: '20%' },
+                { title: 'Subject', align: 'start', key: 'subject', sortable: false, width: '50%' },
+                { title: 'Date Received', align: 'start', key: 'dateReceived', sortable: true, width: '22%' },
+            ],
+        }
+    },
+    computed: {
+        ...mapState(useAuthStore, ['userInfo']),
+        ...mapState(useMessagesStore, ['messages', 'unreadMessageCount']),
+    },
+    created() {
+        this.getNotifications()
+    },
+    methods: {
+        ...mapActions(useMessagesStore, ['getMessages', 'updateMessage']),
+        /**
+         * Initializes the toggle all checkbox and checkbox list state.
+         */
+        initCheckBoxState() {
+            this.checkBoxListState = Array(this.notifications.length).fill(false)
+            this.checkBoxToggleAllState = false
+        },
+        /**
+         * Indicates if the item is read or not.
+         */
+        isRead(item) {
+            return item.selectable.isRead === true
+        },
+        /**
+         * Toggles all the messages checked or unchecked.
+         */
+        toggleAllCheckBoxHandler() {
+            this.checkBoxListState.fill(!this.checkBoxToggleAllState)
+        },
+        /**
+         * Flags a list item has been checked.
+         */
+        checkBoxClickHandler() {
+            this.checkBoxClickedState = true
+        },
+        /**
+         * Handles the row click event. When a row is clicked the message is displayed and marked as read.
+         */
+        rowClickHandler(item, index) {
+            if (!this.checkBoxClickedState) {
+                this.message = item
+                this.rowClickedIndex = index
+                this.updateMessage(item.selectable.messageId, true)
+            } else {
+                this.checkBoxClickedState = false
+            }
+        },
+        /**
+         * Updates all checked messages to read or unread.
+         */
+        updateCheckedReadUnread(isRead) {
+            this.checkBoxListState.forEach((item, index) => {
+                if (item) {
+                    this.updateMessage(this.notifications[index].messageId, isRead)
+                }
+            })
+            this.initCheckBoxState()
+        },
+        /**
+         * Updates a notification to unread.
+         */
+        updateNotificationUnread(item) {
+            if (this.isRead(item)) {
```

Comment:
I don't think this check is necessary. The button only renders when read.

Reply:
Fixed.